docs/source/migrating-from-gateway.mdx

```diff
@@ -36,53 +38,69 @@ The Apollo Router supports most (non-deprecated) `APOLLO_`-prefixed environment
   - `willSendRequest`
   - `didReceiveResponse`
   - `didEncounterError`
-- `logger`
 
-## Specific `plugins` passed to `new ApolloServer({ plugins: [ ... ] })`
+### `logger`
 
-TODO
+The passing of the `logger` option enabled the graph operator to specify a different logger for messages that were produced by the Apollo Gateway.  By default, it inherits from the `logger` on Apollo Server.  This was also useful in order to change the verbosity of logging.
 
-- Talk about the differences in writing plugins in Rust.
-- Link to the life-cycles in the overview of the customizations section (generally link there)
+In the Apollo Router, logging is JSON-structured in production environments by default and the verbosity can be adjusted.  More advanced logging can be enabled through the use of Plugins (discussed later in this article).  For more information on the built-in logging, see our section on [Logging](./configuration/logging).
 
-Apollo Router is trying to minimize the number of cases where custom plugins are necessary to perform common behaviors, for example, setting custom HTTP status codes, transmitting telemetry data, etc.  Out of the box, Apollo Router has a YAML-configuration based approach to propagating headers and support for OpenTelemetry. TODO LINKs
+## Constructor options (including plugins) passed to `new ApolloServer({ ... })`
 
-In general, we're interested in minimizing the need for users to write plugins. In that spirit, prior to writing a plugin, we encourage users to open issues on our GitHub repository with use-cases so we can iterate on possible generalized solutions.
+There are a large variety of uncommon parameters that can be passed to `new ApolloServer({ ... })`, though we'll focus on the most common ones:
 
-That said, some customizations - or time constraints - will likely necessitate some users write plugins.  The Apollo Router's plugins are today written in Rust.  In the future, we expect to offer the ability to write scripted behavior in other languages.  To facilitate writing Rust plugins, we've put together a series of examples of common plugins and those can be found in the Customizations section of our docs.
+- `context`
+- `plugins`
 
-TODO
+To see the full list of options, see [`ApolloServer` options](https://www.apollographql.com/docs/apollo-server/api/apollo-server#options).  If you're using other options, additional migration steps may be necessary to replicate the same behavior.  Please open a discussion on our GitHub repository so we can better understand your needs and hopefully help you with a solution.
+
+### `context`
+
+The `context` parameter is an object that is which TODO
+
+### `plugins`
+
+It was often necessary to write various plugins for Apollo Server, which is the foundation that Apollo Gateway runs on.  These plugins are passed to the `new ApolloServer({ ... })` constructor options in the `plugins` array.  Understanding what each plugin which was enabled will be important to note.
```

Comment:
I can't parse: "Understanding what each plugin which was enabled will be important to note.". Should it be:
"It will be important to understand and document why each plugin was enabled."
?

Reply:
Agree.  And that's the sentiment, yes.  It should probably be more of an extension of the check-list idea I started further up.  How about?:

```suggestion
It was often necessary to write various plugins for Apollo Server, which is the foundation that Apollo Gateway runs on.  These plugins are passed to the `new ApolloServer({ ... })` constructor options in the `plugins` array.  First, take note of each plugin and its use-cases and add it to your check-list, then consider approaches to accomplish the same thing in the Apollo Router.  Plugins might not map one-to-one to plugins in the Apollo Router, but you'll find similar techniques..
```